### PR TITLE
The K2 Migration (feat. The Compose Compiler Gradle Plugin)

### PR DESCRIPTION
**Pull request by emmanuelmuturia**
# Scope
- Migrated to Kotlin version 2.0.0 and upgraded relevant libraries (versions)...
- Added the Compose Compiler Gradle Plugin to the :app, :chai, and :presentation modules...
- Removed the references to the older Compose Compiler...

_Please make sure to read the [Contribution Guidelines](https://github.com/droidconKE/droidconKE2023Android/blob/main/CONTRIBUTING.md)
and check that you understand and have followed it as best as possible Explain what your feature
does in a short paragraph._ please check the below boxes
- [✔] I have followed the coding conventions
- [] I have added/updated necessary tests
- [] I have tested the changes added on a physical device
- [✔] I have run `./codeAnalysis.sh` on linux/unix or `codeAnalysys.bat` on windows to make sure all lint/formatting checks have been done.

## Closes/Fixes Issues
_Declare any issues by typing `fixes #1` or `closes #1` for example so that the automation can kick
in when this is merged_

## Other testing QA Notes
_What have you tested specifically and what possible impacts/areas there are that may need retesting
by others._

Please add a screenshot (if necessary)

### CODE REVIEW

**Comment by chepsi** [comment]
@emmanuelmuturia  I have fixed the failing CI with my latest merge please rebase from develop

**Comment by emmanuelmuturia** [comment]
I have rebased my work as requested...

**Comment by kibettheophilus** [comment]
@emmanuelmuturia I see there are alot of of committed build generated files(the new modules under `shared`), can you please delete then add them to `.gitignore`?

**Comment by emmanuelmuturia** [comment]
@kibettheophilus I have rewritten my commit history and added the build-generated files to `.gitignore` as advised...

**Comment by kibettheophilus** [comment]
> @kibettheophilus I have rewritten my commit history and added the build-generated files to `.gitignore` as advised...



> @kibettheophilus I have rewritten my commit history and added the build-generated files to `.gitignore` as advised...



> @kibettheophilus I have rewritten my commit history and added the build-generated files to `.gitignore` as advised...

I see there are still generated files, can you delete all the changes you have `shared/.....`

**Comment by emmanuelmuturia** [comment]
@kibettheophilus Please check if my latest push matches your recommendation...

**Review by kibettheophilus — APPROVED** [review]
LGTM

File: app/build.gradle.kts [hunk 1]
```diff
@@ -16,6 +16,7 @@
 plugins {
     alias(libs.plugins.droidconke.android.application)
     alias(libs.plugins.droidconke.android.hilt)
+    alias(notation = libs.plugins.compose.compiler)
```
Comment (chepsi) [inline app/build.gradle.kts:19]: why is this named?

File: app/src/main/res/values/strings.xml [hunk 2]
```diff
@@ -1,3 +1,3 @@
 <resources>
-    <string name="app_name">DroidconKE2023</string>
+    <string name="app_name">DroidConKE</string>
```
Comment (chepsi) [inline app/src/main/res/values/strings.xml:2]: 🏅

Comment (chepsi) [reply app/src/main/res/values/strings.xml:2]: `DroidConKE` -> `DroiconKe`

File: chai/build.gradle.kts [hunk 3]
```diff
@@ -16,6 +16,7 @@
 plugins {
     alias(libs.plugins.droidconke.android.library)
     alias(libs.plugins.droidconke.android.library.compose)
+    alias(notation = libs.plugins.compose.compiler)
```
Comment (chepsi) [inline chai/build.gradle.kts:19]: Minor: Why is this named?

**Review by chepsi — APPROVED** [review]
LGTM

**Review by kibettheophilus — CHANGES_REQUESTED** [review]
Hey @emmanuelmuturia your latest push added the build files, please check again

**Comment by emmanuelmuturia** [comment]
@chepsi I have implemented the minor corrections as suggested...

**Comment by emmanuelmuturia** [comment]
@kibettheophilus @chepsi Please confirm if everything is all good now...

File: app/src/main/res/values/strings.xml [hunk 4]
```diff
@@ -1,3 +1,3 @@
 <resources>
-    <string name="app_name">DroidconKE2023</string>
+    <string name="app_name">Droidconke</string>
```
Comment (kibettheophilus) [inline app/src/main/res/values/strings.xml:2]:
```suggestion
    <string name="app_name">DroidconKe</string>
```

**Review by kibettheophilus — APPROVED** [review]

**Comment by kibettheophilus** [comment]
> @kibettheophilus @chepsi Please confirm if everything is all good now...

looks good now

**Comment by etonotieno** [comment]
K2 adds [a new `.kotlin` directory](https://kotlinlang.org/docs/whatsnew20.html#new-directory-for-kotlin-data-in-gradle-projects) for Gradle projects which should be ignored by git.

@kibettheophilus @chepsi

**Comment by kibettheophilus** [comment]
> K2 adds [a new `.kotlin` directory](https://kotlinlang.org/docs/whatsnew20.html#new-directory-for-kotlin-data-in-gradle-projects) for Gradle projects which should be ignored by git.
> 
> image
> 
> @kibettheophilus @chepsi

@etonotieno correct, let me add it

**Comment by kibettheophilus** [comment]
@etonotieno review https://github.com/droidconKE/droidconKeKotlin/pull/300

**Pull request merged**